src/perception/tag_detector/long_range_cam/long_range_tag_detector.processing.cpp

```diff
@@ -0,0 +1,214 @@
+                } else {
+                    it++;
+                }
+            }
+        }
+
+
+        //decrement non updated & set updated status to false
+    }
+
+    LongRangeTagStruct LongRangeTagDetectorNodelet::createLrt(int tagId, std::vector<cv::Point2f>& tagCorners) {
+        //TODO: Modify the LRT struct and this function to assign a bearing instead of an imageCenter
+        LongRangeTagStruct lrt;
+
+        lrt.hitCount = mBaseHitCount; //Start at base hit count value
+        lrt.id = tagId;
+        lrt.updated = true;
+
+        lrt.imageCenter = getNormedTagCenterOffset(tagCorners);
+        std::cout << "lrt image center " << lrt.imageCenter.x << std::endl;
+
+        return lrt;
+    }
+
+    void LongRangeTagDetectorNodelet::updateNewlyIdentifiedTags(size_t tagIndex) {
+        int currentId = mImmediateIds[tagIndex];
+
+        //Create new struct for each tag
+        LongRangeTagStruct lrt = createLrt(currentId, mImmediateCorners[tagIndex]);
+
+        //Check if the tag was already detected and update hitCount to reflect
+        if (mTags.contains(currentId)) {
+            //Key exist sin mTags
+            lrt.hitCount = std::min(mTags[currentId].hitCount + mTagIncrementWeight, mMaxHitCount);
+        }
+        mTags[currentId] = lrt;
+    }
+
+    cv::Point2f LongRangeTagDetectorNodelet::getTagCenterPixels(std::vector<cv::Point2f>& tagCorners) {
+        cv::Point2f centerPoint;
+        float centerX = 0;
+        float centerY = 0;
+
+        for (size_t i = 0; i < 4; i++) {
+            centerX += tagCorners[i].x;
+            centerY += tagCorners[i].y;
+        }
+
+        centerX /= 4.0;
+        centerY /= 4.0;
+
+        centerPoint.x = centerX;
+        centerPoint.y = centerY;
+
+        return centerPoint;
+    }
+
+    cv::Point2f LongRangeTagDetectorNodelet::getTagCenterOffsetPixels(std::vector<cv::Point2f>& tagCorners) const {
+        cv::Point2f centerPoint = getTagCenterPixels(tagCorners);
+
+        centerPoint.x -= static_cast<float>(mImgMsg.width);
+
+        //-1 is necessary because of 0,0 being in the top left
+        centerPoint.y = static_cast<float>(-1.0) * (centerPoint.y - static_cast<float>(mImgMsg.height));
+
+        return centerPoint;
+    }
+
+    cv::Point2f LongRangeTagDetectorNodelet::getNormedTagCenterOffset(std::vector<cv::Point2f>& tagCorners) const {
+        cv::Point2f offsetCenterPoint = getTagCenterOffsetPixels(tagCorners);
+
+        offsetCenterPoint.x /= static_cast<float>(mImgMsg.width);
+        offsetCenterPoint.y /= static_cast<float>(mImgMsg.height);
+
+        return offsetCenterPoint;
+    }
+
+    float LongRangeTagDetectorNodelet::getTagBearing(cv::Point2f& tagCenter) const {
+        //for HD720 resolution
+        auto imageWidth = (float) mImgMsg.width;
+        std::cout << "width: " << imageWidth << " tag center x: " << tagCenter.x << std::endl;
+        float bearing = -1 * ((float) tagCenter.x + 0.5) * mLongRangeFov;
```

Comment:
why +0.5?

Reply:
this is bc tagCenter is a normed center [0,1] of the tag's location on the image. This should prob be changed not very intuitive